group-size-refinement-list/react-instantsearch/src/lib/GroupSizeRefinementList.js

```diff
@@ -0,0 +1,57 @@
+import React, { Component } from "react";
+import { Highlight, connectRefinementList } from "react-instantsearch-dom";
+
+import "./GroupSizeRefinementList.css";
+
+class GroupSizeRefinementList extends Component {
+    render() {
+        let sizeGroups = [];
+        let inclusionArrays = [];
+        if (this.props.patterns) {
+            this.props.patterns.forEach((regex, i) => {
```

Comment:
this could be done with map instead of push in the forEach

Reply:
Not really sure about that, I need `inclusionArrays` to be update on each iteration to perform
a filter with `!inclusionArrays.includes(hit)` (used to remove duplicates)

If I do this: 
```
            inclusionArrays = this.props.patterns.map((regex, i) => {
                sizeGroups[i] = this.props.items.filter(
                    hit => regex.test(hit.label) && !inclusionArrays.includes(hit)
                );
                return [...sizeGroups[i]]
            });
```
I will get duplicates because !inclusionArrays.includes(hit) doesn't have any effect

Reply:
Ah yes, in that case you'd need a reduce which might be less readable than this